### PR TITLE
Add support for experimental flags during hot reload.

### PR DESCRIPTION
**Pull request by a-siva**

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -347,6 +353,7 @@ class RunCommand extends RunCommandBase {
         fileSystemRoots: argResults['filesystem-root'],
         fileSystemScheme: argResults['filesystem-scheme'],
         viewFilter: argResults['isolate-filter'],
+	experimentalFlags: expFlags,
```
Comment (bkonyi) [inline packages/flutter_tools/lib/src/commands/run.dart]: dartfmt probably won't like this.

**Review by bkonyi — APPROVED** [review]
LGTM with formatting nit.

**Review by aam — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/run.dart (hunk 1 above)
Comment (a-siva) [reply packages/flutter_tools/lib/src/commands/run.dart]: Replaced tab with spaces.

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 2]
```diff
@@ -339,6 +339,12 @@ class RunCommand extends RunCommandBase {
       throwToolExit('Error: --train is only allowed when running as --dynamic --profile '
           '(recommended) or --debug (may include unwanted debug symbols).');
 
+    List<String> expFlags;
+    if (argParser.options.containsKey(FlutterOptions.kEnableExperiment) &&
+        argResults[FlutterOptions.kEnableExperiment] != null) {
```
[the remark this answers is not in the document]
Comment (a-siva) [reply packages/flutter_tools/lib/src/commands/run.dart]: You would prefer
argResults[FlutterOptions.kEnableExperiment].isNotEmpty

**Pull request merged**